partiql-eval/src/main/kotlin/org/partiql/eval/internal/operator/rel/RelDistinct.kt

```diff
@@ -3,16 +3,19 @@ package org.partiql.eval.internal.operator.rel
 import org.partiql.eval.internal.Environment
 import org.partiql.eval.internal.Record
 import org.partiql.eval.internal.operator.Operator
+import org.partiql.value.ListValue
 import org.partiql.value.PartiQLValue
 import org.partiql.value.PartiQLValueExperimental
+import org.partiql.value.listValue
+import java.util.TreeSet
 
 internal class RelDistinct(
     val input: Operator.Relation
 ) : RelPeeking() {
 
-    // TODO: Add hashcode/equals support for PQLValue. Then we can use Record directly.
+    // TODO: Add hashcode/equals support for Datum. Then we can use Record directly.
     @OptIn(PartiQLValueExperimental::class)
-    private val seen = mutableSetOf<List<PartiQLValue>>()
+    private val seen = TreeSet<ListValue<PartiQLValue>>(PartiQLValue.comparator())
```

Comment:
Hmm is hashcode/equals support for Datum tracked anywhere?

Would be nice to replace this `TreeSet` implementation using a custom comparator with `Datum` before `v1` is released.

Reply:
Yeah, PartiQLValue is slowly being ripped out of the evaluator. Since it's an internal implementation detail, mind if we tackle this when it's fully ripped out?

Reply:
(from offline discussion) yeah, seems reasonable to tackle this TODO later since it will affect a lot of other parts of the evaluator (e.g. bag ops).

might be worthwhile to cut an issue to track